chain/epoch_manager/src/reward_calculator.rs

```diff
@@ -25,18 +26,29 @@ impl RewardCalculator {
         validator_block_chunk_stats: HashMap<AccountId, BlockChunkValidatorStats>,
         validator_stake: &HashMap<AccountId, Balance>,
         total_supply: Balance,
+        protocol_version: ProtocolVersion,
     ) -> (HashMap<AccountId, Balance>, Balance) {
         let mut res = HashMap::new();
         let num_validators = validator_block_chunk_stats.len();
-        let epoch_total_reward = (U256::from(*self.max_inflation_rate.numer() as u64)
+        let max_inflation_rate = if protocol_version >= ENABLE_INFLATION_PROTOCOL_VERSION {
```

Comment:
Is the plan to keep the logic with the constant 1/20 forever, or to eventually remove that logic and just keep self.max_inflation_rate?

Generally, I prefer the construction in which the genesis protocol version is known, and that then looks like this:

```if genesis_protocol_version < epoch_protocol_version && epoch_protocol_version >= UPGRADE_VERSION {
    <the new value hardcoded>
} else {
    <value from the genesis config>
}```

(But doing it like it is done, with the plan in the medium term to change the OLDEST_SUPPORTED_VERSION to the version in which the inflation was enabled and remove the hard coded constant is also fine).

Reply:
> Is the plan to keep the logic with the constant 1/20 forever, or to eventually remove that logic and just keep self.max_inflation_rate?

Do I understand correctly that `max_inflation_rate` is coming from the current genesis (which is currently `0`)? This means unless we rebase genesis via hardfork we need to keep current genesis and this code so that people can  re-validate all blocks from the genesis, if needed (e.g. this is needed for Rosetta).

Reply:
Yes, but that is not really in conflict with what Alex suggested. Constructing with genesis protocol version allows better flexibility for networks other than mainnet (which may experience hardforks due to various reasons).